File: solved.ac/graph_traversal/10159.py
import sys
input = sys.stdin.readline
n = int(input())
m = int(input())
a = [[0] * n for _ in range(n)]
for _ in range(m):
  x, y = map(int, input().split())
  a[x-1][y-1] = 1
for k in range(n):
  for i in range(n):
    for j in range(n):
      if a[i][k] and a[k][j]:
        a[i][j] = 1
for i in range(n):
  cnt = 0
  for j in range(n):
    if not a[i][j] and not a[j][i]:
      cnt += 1
  print(cnt -1)

File: solved.ac/graph_traversal/10159.js
const fs = require("fs")
let input = fs.readFileSync("./dev/stdin").toString().trim().split("\n")
let [n, m] = input
const graph = Array.from({ length: +n + 1 }, () =>
  Array.from({ length: +n + 1 }, () => 0)
)
const edges = input.slice(2).map((x) => x.split(" ").map((x) => +x))
for ([a, b] of edges) {
  graph[a][b] = 1
}
for (let k = 1; k <= n; k++) {
  for (let i = 1; i <= n; i++) {
    for (let j = 1; j <= n; j++) {
      if (graph[i][k] && graph[k][j]) {
        graph[i][j] = 1
      }
    }
  }
}
for (let i = 1; i <= n; i++) {
  let cnt = 0
  for (let j = 1; j <= n; j++) {
    if (!graph[i][j] && !graph[j][i]) cnt++
  }
  console.log(cnt - 1)
}
